main.js

```diff
@@ -0,0 +1,122 @@
+//Fix problem with deleting comments and adding new comments
+
+//Fix problem with comments section showing comments for every post no matter what post was clicked
+
+
+
+//create list of posts and names to be associated with those posts
+var postsArray = [];
+
+//adding each post to list of posts
+var postsObj = function(msg, name) {
+  postsArray.push({
+    userMsg: msg,
+    userName: name,
+    post: postsArray.length + 1,
+    comments: []
+  })
+};
+
+//Allows user to post upon clicking 'post'
+$('#post').on('click', function() {
+  postsObj($('#message').val(), $('#name').val());
+  console.log(postsArray[postsArray.length - 1].post);
+  //console.log(postsArray);
+  var newPost = '<div><div><button type="button" style="padding:0px" class="btn-link remove">remove</button><button type="button" class="btn-link comments" id="' + postsArray[postsArray.length - 1].post + '" data-toggle="modal" data-target="commentsModal">comments</button></div><div class="msg">' + postsArray[postsArray.length - 1].userMsg + '</div><div class="name">Posted by: <b>' + postsArray[postsArray.length - 1].userName + '</b></div><hr></div>'
+  $('.posts').append(newPost);
+
+  //clear text boxes after each post
+  $('#message').val('');
+  $('#name').val('');
+
+  //remove posts upon clicking 'remove' button
+  $('.remove').on('click', function() {
+    $(this).parent().parent().remove();
+  });
+
+  //show comments modal when user clicks 'comments' button
+  $('.comments').on('click', function(e) {
+    e.stopImmediatePropagation();
+    $('#comments-modal').css({"display":"block"});
+
+    //comments modal to feature original post
+    var id = $(this).attr('id');
+    var postModal = '<div class="show-post" id="' + '"><div class="msg">' + postsArray[id - 1].userMsg + '</div><div class="name">Posted by: <b>' + postsArray[id - 1].userName + '</b></div><hr></div>'
+    $('.post-modal').append(postModal);
+
+    //adding each comment to list of posts
+    var commentObj = function(msg, name) {
+      postsArray[id - 1].comments.push({
```

Comment:
This `id` thing is good, but it's technically the same thing as the index, so you could just use that.

The best thing to do would have been to render your posts and comments based on this great data structure you've created - `postsArray`